compiler/scenario-service/client/src/DA/Daml/LF/PrettyScenario.hs

```diff
@@ -797,6 +806,18 @@ prettyNodeNode nn = do
         ppChildren <- prettyChildren node_RollbackChildren
         pure $ keyword_ "rollback" $$ ppChildren
 
+-- | Take a list of parties and the singular and multiple present tense verbs
+-- Depending on the count of parties, returns a prettified list of these elements, using @,@ and @and@, as well as the correct verb
+-- e.g. @[a, b, c] -> a, b and c@
+partiesAction :: V.Vector Party -> String -> String -> (Doc SyntaxClass, Doc SyntaxClass)
+partiesAction pv singular multiple = go $ mapV prettyParty pv
+  where
+    go [] = (text "No-one/unknown", keyword_ singular)
+    go [p] = (p, keyword_ singular)
+    go ps = (fcommasep init <-> keyword_ "and" <-> head last, keyword_ multiple)
+      where
+        (init, last) = splitAt (length ps - 1) ps
```

Comment:
here I'd use [`unsnoc`](https://hackage.haskell.org/package/extra-1.7.12/docs/Data-List-Extra.html#v:unsnoc) since this [already depends on `extra`](https://github.com/digital-asset/daml/blob/akrmn/dynamic-exercise/compiler/scenario-service/client/BUILD.bazel#L21).

Reply:
I can see that, though that Maybe could get messy without a fromJust